internal/core/config.go

```diff
@@ -19,10 +19,54 @@
 package core
 
 import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
 	"github.com/sapcc/castellum/internal/db"
 )
 
 // Config contains the app-level configuration options.
 type Config struct {
-	MaxAssetSize map[db.AssetType]*uint64
+	MaxAssetSizeRules []MaxAssetSizeRule
+}
+
+type MaxAssetSizeRule struct {
+	Regex *regexp.Regexp
+	Value *uint64
+}
+
+func (c Config) MaxAssetSizeFor(AssetType db.AssetType) (maxAssetSize *uint64) {
+	for _, rule := range c.MaxAssetSizeRules {
+		if rule.Regex.MatchString(string(AssetType)) {
+			if maxAssetSize == nil || *rule.Value < *maxAssetSize {
+				maxAssetSize = rule.Value
+			}
+		}
+	}
+
+	return maxAssetSize
+}
+
+func (c *Config) SetMaxAssetSizeRules(maxAssetSizeString string) error {
+	maxAssetSizes := strings.Split(maxAssetSizeString, ",")
+	for _, v := range maxAssetSizes {
+		sL := strings.Split(v, "=")
+		if len(sL) != 2 {
+			return fmt.Errorf("expected a max asset size configuration value in the format: '<asset-type>=<max-asset-size>', got: %s", v)
+		}
+		assetType := sL[0]
+		maxSize, err := strconv.ParseUint(sL[1], 10, 64)
+		if err != nil {
+			return err
+		}
+
+		rgx := regexp.MustCompile("^" + assetType + "$")
```

Comment:
Since this regex is supplied by the user, we should not panic on parse error. Please use `regexp.Compile()` instead.